### PR TITLE
[New] no-unused-modules: support dynamic imports [2]

### PR DESCRIPTION
**Pull request by Hypnosphi**
Replaces #1660

### CODE REVIEW

**Comment by ljharb** [comment]
oof, please don't make new PRs (but please leave this one open, since now I have to keep it in sync with #1660 until both are merged)

**Comment by Hypnosphi** [comment]
Sorry, the scheme with two branches for one PR apparently didn't work well

**Comment by ljharb** [comment]
Probably something that landed in master in the interim, since the branches shouldn't be any different.

**Comment by Hypnosphi** [comment]
My branch is up-to date with master

**Comment by Hypnosphi** [comment]
All green now

File: .github/workflows/node-4+.yml [hunk 1]
```diff
@@ -81,6 +81,7 @@ jobs:
           node-version: ${{ matrix.node-version }}
           after_install: npm run copy-metafiles && ./tests/dep-time-travel.sh
           skip-ls-check: true
+      - run: npm run pretest
```
Comment (ljharb) [inline .github/workflows/node-4+.yml:84]: why is this needed here? pretest is `linklocal` which is meant only for local development.

Comment (Hypnosphi) [reply .github/workflows/node-4+.yml:84]: How else can I link the yet unreleased version of `eslint-plugin-utils` for tests?

Comment (Hypnosphi) [reply .github/workflows/node-4+.yml:84]: BTW `appveyor.yml` includes this script as well:
https://github.com/import-js/eslint-plugin-import/blob/main/appveyor.yml#L88

Comment (Hypnosphi) [reply .github/workflows/node-4+.yml:84]: An example of a failure: https://github.com/import-js/eslint-plugin-import/runs/3464305918

Comment (ljharb) [reply .github/workflows/node-4+.yml:84]: Fair enough; I guess we don't change the utils package often enough for this to have been a problem.

**Comment by ljharb** [comment]
post rebase, travis-ci is still failing, with `Error: Cannot find module 'eslint-module-utils/visit'` - despite https://github.com/import-js/eslint-plugin-import/commit/968dda15485e85a2bc432f025a6f7d1cbe532ffd (which wasn't required for travis tests to be passing before)

**Comment by Hypnosphi** [comment]
For some reason, I don't see `npm run pretest` in the job log: https://app.travis-ci.com/github/import-js/eslint-plugin-import/jobs/534928091

**Comment by ljharb** [comment]
hm, you're right, i don't see that either. Maybe `after_install` isn't a thing, i'll fix that.

**Review by ljharb — APPROVED** [review]

**Comment by aladdin-add** [comment]
👍 awesome! thx for fixing this!

**Comment by Meemaw** [comment]
Hey, is it possible to release this? 🙏
